pkg/controllers/inventory/inventory_controller.go

```diff
@@ -126,25 +126,25 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	// Watch for changes to SyncSetInstances
+	// Watch for changes to ClusterSync
 	err = c.Watch(
-		&source.Kind{Type: &hivev1.SyncSetInstance{}},
+		&source.Kind{Type: &hiveinternalv1alpha1.ClusterSync{}},
 		&handler.EnqueueRequestsFromMapFunc{
 			ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
-				syncSetInstance, ok := a.Object.(*hivev1.SyncSetInstance)
+				clusterSync, ok := a.Object.(*hiveinternalv1alpha1.ClusterSync)
 				if !ok {
-					// not a SyncSetInstance, returning empty
-					klog.Error("SyncSetInstance handler received non-SyncSetInstance object")
+					// not a ClusterSync, returning empty
+					klog.Error("ClusterSync handler received non-ClusterSync object")
 					return []reconcile.Request{}
 				}
 				bmas := &inventoryv1alpha1.BareMetalAssetList{}
-				err := mgr.GetClient().List(context.TODO(), bmas, client.InNamespace(syncSetInstance.Namespace))
+				err := mgr.GetClient().List(context.TODO(), bmas, client.InNamespace(clusterSync.Namespace))
 				if err != nil {
 					klog.Error("Could not list BareMetalAssets", err)
 				}
 				var requests []reconcile.Request
 				for _, bma := range bmas.Items {
-					if bma.Spec.ClusterDeployment.Name == syncSetInstance.Spec.ClusterDeploymentRef.Name {
+					if bma.Spec.ClusterDeployment.Name == clusterSync.Name {
 						requests = append(requests, reconcile.Request{
 							NamespacedName: types.NamespacedName{
 								Name:      bma.Name,
```

Comment:
is this match correct? clustersync's name is same as bma's name?

Reply:
from my understanding, clusterSync can have many syncsets?  but wondering how will you get bma name in that case 😂

Reply:
nvm, this looks good, so clustersync name is same as clusterdeployment name 👍